master-firmware/src/dynamic_obstacle_avoidance.c

```diff
@@ -0,0 +1,47 @@
+#include <ch.h>
+
+#include <msgbus/messagebus.h>
+#include <error/error.h>
+#include "priorities.h"
+#include "base/base_controller.h"
+#include "main.h"
+
+#include "dynamic_obstacle_avoidance.h"
+
+#define DYNAMIC_OBSTACLE_AVOIDANCE_STACKSIZE 128
+
+typedef struct {
+    float distance;
+    float angle;
+} beacon_signal_t;
+
+
+void dynamic_obstacle_avoidance_policy(void* _robot)
```

Comment:
I am not too sure about having the dynamic obstacle avoidance policy in another thread from strategy is a wise choice, but I think it could be easy enough to merge in the `trajectory_test_end` function, like Microb Technology did, if we need to later on